### ISSUE TITLE
[TCGC] Follow ups on the behavior of TCGC on `@format` decorator

### ISSUE THREAD

**Issue by ArcturusZhang** [Azure/typespec-azure#178]
Per discussion, 

> Those formats are a thing in openapi because everything needs to be a primitive type but we don’t have this limitation in typespec so why are we digging our self back into this mess.

Therefore maybe we should not parse the content of `@format` in TCGC any more (we even throw errors when the content of format is not recognized).
But we still have a hard requirement of representing some azure-specific types, such as etag, azure-location, arm-id.

Therefore I believe we should have the following follow ups:
- [ ] TCGC should stop parsing `string` with some specific `@format` into different built-in types.
- [ ] New built-in scalar types should be introduced in azure-specific templates
	- [x] `arm-id`: https://github.com/Azure/typespec-azure/blob/976e838d3200d824e8407f959054a557eead47a5/packages/typespec-azure-resource-manager/lib/models.tsp#L108
	- [x] `etag`: https://github.com/Azure/typespec-azure/blob/976e838d3200d824e8407f959054a557eead47a5/packages/typespec-azure-core/lib/models.tsp#L345
	- [ ] `azure-location`: I did not find a definition for azure-location
	- [x] `ip-address`:
	https://github.com/Azure/typespec-azure/blob/976e838d3200d824e8407f959054a557eead47a5/packages/typespec-azure-core/lib/models.tsp#L314
	https://github.com/Azure/typespec-azure/blob/976e838d3200d824e8407f959054a557eead47a5/packages/typespec-azure-core/lib/models.tsp#L329
- [ ] Our MPG swagger-typespec converter should include information like this into the typespec

**Comment by lirenhe** [comment]
For Azure location, it is scheduled for Feb.  https://github.com/Azure/typespec-azure-pr/issues/3064

**Comment by iscai-msft** [comment]
@ArcturusZhang after talking with @timotheeguerin, we have 2 changes
1. We keep the different string types (i.e. `ipAddress`, `uri` etc) and translate 1-1 from TypeSpec. We only set scalars if they are from the expected namespace (i.e. `TypeSpec` or `Azure.Core`
2. We will continue to parse format into kinds for legacy reasons, but we also verify namespace as above, and this is more legacy code

Closing because this issue has been addressed
